build/builder.go

```
@@ -451,11 +451,16 @@ func (b *Builder) deleteRemainingShards() {
 		shard := b.nextShardNum
 		b.nextShardNum++
 		name := b.opts.shardName(shard)
-		if err := os.Remove(name); os.IsNotExist(err) {
+		// best effort: we get an error paths is empty and we assume there is
+		// nothing more to cleanup.
+		paths, _ := zoekt.IndexFilePaths(name)
+		if len(paths) == 0 {
 			break
-		} else {
-			b.shardLog("remove", name)
 		}
+		for _, p := range paths {
+			_ = os.Remove(p)
```

Should we log if os.Remove fails?

---

The previous code path did not log, was best effort. Same reason we ignore the error above. I'm happy adding in logging though. Will do that.